modules/core/src/v2/baseCoin.ts

```diff
@@ -330,22 +393,21 @@ export abstract class BaseCoin {
 
       const isKrsRecovery = backupKey.startsWith('xpub') && !userKey.startsWith('xpub');
 
-      const validatePassphraseKey = function(userKey, passphrase) {
+      function validatePassphraseKey(userKey: string, passphrase?: string): bitcoin.HDNode {
         try {
           if (!userKey.startsWith('xprv') && !userKey.startsWith('xpub')) {
             userKey = self.bitgo.decrypt({
               input: userKey,
               password: passphrase,
             });
           }
-          const userHDNode = bitcoin.HDNode.fromBase58(userKey);
-          return Promise.resolve(userHDNode);
+          return bitcoin.HDNode.fromBase58(userKey);
         } catch (e) {
           throw new Error('Failed to decrypt user key with passcode - try again!');
         }
-      };
+      }
 
-      const key = yield validatePassphraseKey(userKey, passphrase);
+      const key: bitcoin.HDNode = validatePassphraseKey(userKey, passphrase);
```

Comment:
no more ```yield``` ?

Reply:
nope. `validatePassphraseKey` is a function which is internal to this outer function, and it never needed to be async in the first place.